sdv/constraints/tabular.py

```diff
@@ -98,7 +102,8 @@ def _valid_separator(self, table_data):
                 Whether the separator is valid for this data or not.
         """
         for column in self._columns:
-            if table_data[column].str.contains(self._separator).any():
+            if table_data.dtypes[column] == object and \
```

Comment:
This has been changed in the `master` branch, so I believe this change is not necessary any more.